tests/test_paperqa.py

```diff
@@ -1199,3 +1199,71 @@ def test_answer_rename(recwarn) -> None:
 def test_dois_resolve_to_correct_journals(doi_journals):
     details = DocDetails(doi=doi_journals["doi"])  # type: ignore[call-arg]
     assert details.journal == doi_journals["journal"]
+
+
+@pytest.mark.vcr
+@pytest.mark.parametrize("use_partition", [True, False])
+@pytest.mark.asyncio
+async def test_partitioning_fn_docs(use_partition: bool) -> None:
+    settings = Settings.from_name("fast")
+    settings.answer.evidence_k = 2  # limit to only 2
+
+    # imagine we have some special selection we want to
+    # embedding rank by itself
+    def partition_by_citation(doc: Doc) -> int:
+        if "special" in doc.citation:
+            return 1
+        return 0
+
+    partitioning_fn = None
+
+    if use_partition:
+        partitioning_fn = partition_by_citation
```

Comment:
```suggestion
    partitioning_fn = partition_by_citation if use_partition else None
```

Let's do it in one line for our dear friend `mypy`